packages/config/test/load-config.test.js

```diff
@@ -0,0 +1,302 @@
+    t.teardown(() => {
+      process.chdir(cwd)
+    })
+  }
+  const file = join(__dirname, 'fixtures', 'platformatic.service.json')
+  const { configManager, args } = await loadConfig({}, [], app)
+
+  t.match(args, {
+    _: [],
+    allowEnv: '',
+    'allow-env': '',
+    E: ''
+  })
+  t.same(configManager.current, JSON.parse(await readFile(file, 'utf8')))
+})
+
+test('empty rejects with an error', async t => {
+  t.plan(2)
+  const cwd = process.cwd()
+  process.chdir(join(__dirname, 'fixtures', 'empty'))
+  t.teardown(() => {
+    process.chdir(cwd)
+  })
+
+  try {
+    await loadConfig({}, [], app)
+  } catch (err) {
+    t.equal(err.message, 'no config file found')
+    t.same(err.filenames, [
+      'platformatic.service.json',
+      'platformatic.service.json5',
+      'platformatic.service.yaml',
+      'platformatic.service.yml',
+      'platformatic.service.toml',
+      'platformatic.service.tml',
+      'platformatic.json',
+      'platformatic.json5',
+      'platformatic.yaml',
+      'platformatic.yml',
+      'platformatic.toml',
+      'platformatic.tml'
+    ])
+  }
+})
+
+test('not passing validation kills the process', async t => {
+  t.plan(2)
+  function app () {
+  }
+  app.configType = 'service'
+  app.schema = {
+    $id: 'service',
+    type: 'object',
+    properties: {
+      foo: {
+        type: 'string'
+      }
+    },
+    required: ['foo']
+  }
+
+  app.configManagerConfig = {
+    schema: app.schema
+  }
+
+  const file = join(__dirname, 'fixtures', 'platformatic.service.json')
+  try {
+    await loadConfig({}, ['-c', file, '--boo'], app)
+    console.log('pass')
+  } catch (err) {
+    t.equal(err.message, 'The configuration does not validate against the configuration schema')
+    t.same(err.validationErrors, [{
+      path: '/',
+      message: 'must have required property \'foo\' {"missingProperty":"foo"}'
+    }])
+  }
+})
+
+test('allow-env', async (t) => {
+  const file = join(__dirname, 'fixtures', 'bad-placeholder.json')
+  process.env.PORT = '3000'
+  const { configManager, args } = await loadConfig({}, ['-c', file, '--allow-env', 'PORT'], app)
+
+  t.same(args, {
+    _: [],
+    c: file,
+    config: file,
+    allowEnv: 'PORT',
+    'allow-env': 'PORT',
+    E: 'PORT'
+  })
+  const content = JSON.parse(await readFile(file, 'utf8'))
+  content.server.port = '3000'
+  t.same(configManager.current, content)
+})
+
+test('loadConfig with Store', async t => {
+  const file = join(__dirname, 'fixtures', 'with-store.json')
+
+  const store = new Store()
+  store.add(app)
+  const { configManager, args } = await loadConfig({}, ['-c', file, '--boo'], store)
+
+  t.same(args, {
+    _: [],
+    c: file,
+    config: file,
+    boo: true,
+    allowEnv: '',
+    'allow-env': '',
+    E: ''
+  })
+  t.same(configManager.current, JSON.parse(await readFile(file, 'utf8')))
+})
+
+test('printConfigValidationErrors', async t => {
+  t.plan(1)
+  const table = console.table
+  console.table = (data) => {
```

Comment:
Just throwing this out there: you could use `mock` from `node:test` here if you wanted while still using tap to run the tests in this file.